File: plasma/8_playground/calc/electric_potential.f90
module electric_potential
    use pic_utils
    use numerical_quantity
    use physical_quantity
    implicit none
contains
    function calc_electric_potential_with_no_collision(ion_density)
        real, intent(in) :: ion_density(LATTICE_NUM, LATTICE_NUM, LATTICE_NUM)
        real :: calc_electric_potential_with_no_collision(LATTICE_NUM, LATTICE_NUM, LATTICE_NUM)
        real :: k, T, e, n0
        k = BOLTZMANN_CONSTANT
        T = ELECTRON_TEMPATATURE
        e = ELEMENTARY_CHARGE
        n0 = PLASMA_DENSITY
        calc_electric_potential_with_no_collision = -k * T / e &
        * log(1+ion_density / n0)
    end function
end module electric_potential

File: plasma/6_playground/calc/electric_potential.py
import numpy as np
import math
import pathlib
import sys

root_dir = pathlib.Path(__file__).resolve().parent.parent
sys.path.append(str(root_dir))
from utils.pic_module import PICModule
from utils.random_module import RandomModule
from models import physical_quantity as pq


class ElectricPotential:

    """電位を定義"""
    # TODO potentialの計算式どうする？
    
    def __init__(self):
        pass
        
    def calc_electric_potential_with_no_collision(
        self,
        ion_density: np.array
    ) -> np.array:
        k = pq.BOLTZMANN_CONSTANT
        T = pq.ELECTRON_TEMPATATURE
        e = pq.ELEMENTARY_CHARGE
        n0 = pq.PLASMA_DENSITY
        #  ここiondensityが0の時どうするの？
        potential = - k * T * np.log1p(ion_density / n0) / e
        return potential

def main():
    potential = ElectricPotentialModel()


if __name__ == "__main__":
    main()